libs/math/include/math/tensor.hpp

```diff
@@ -645,22 +647,23 @@ class Tensor
     using Type = typename STensor::Type;
 
     TensorSliceImplementation<STensor>(STensor &t, std::vector<std::vector<SizeType>> range,
-                                       SizeType axis = 0)
+                                       std::vector<SizeType> axis)
```

Comment:
Would one option for efficiency be to have an axis_ and an axes_ member variable (could even be a union if we're really fancy); then have both constructors: one for SizeType axis = 0, and one for SizeVector axes?